code/TagField.php

```diff
@@ -173,4 +273,90 @@ public function saveInto(DataObjectInterface $record) {
 			$record->$name = implode(',', $values);
 		}
 	}
+
+	/**
+	 * Returns a JSON string of tags, for ajax-based search.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return string
+	 */
+	public function suggest(SS_HTTPRequest $request) {
+		if($this->recordClass === null) {
+			return '[]';
+		}
+
+		/**
+		 * @var DataObject $object
+		 */
+		$object = singleton($this->recordClass);
+
+		$term = Convert::raw2sql($request->getVar('term'));
+
+		$tags = array();
+
+		if($object->hasMethod($this->name)) {
+			$tags = $this->getObjectTags($this->recordClass, $object, $term);
+		} elseif($object->hasField($this->name)) {
+			$tags = $this->getStringTags($this->recordClass, $object, $term);
+		}
+
+		return Convert::raw2json($tags);
+	}
+
+	/**
+	 * @param string $class
+	 * @param DataObject $instance
+	 * @param string $term
+	 *
+	 * @return array
+	 */
+	protected function getObjectTags($class, DataObject $instance, $term) {
+		$relation = $instance->{$this->name}();
+
+		$query = DataList::create($relation->dataClass())
+			->where('"' . $this->relationTitle . '" LIKE \'%' . $term . '%\'')
+			->sort($this->relationTitle)
+			->limit($this->limit);
+
+		$items = array();
+
+		foreach($query as $record) {
+			$items[] = $record->{$this->relationTitle};
```

Comment:
If just selecting one field, use `->column($this->relationTitle)` to only limit the SQL query to that one field.